packages/browser/src/login/oidc/oidcHandlers/AuthorizationCodeWithPkceOidcHandler.ts

```diff
@@ -59,36 +61,57 @@ export default class AuthorizationCodeWithPkceOidcHandler
   }
 
   async handle(oidcLoginOptions: IOidcOptions): Promise<void> {
-    const requestUrl = new URL(
-      oidcLoginOptions.issuerConfiguration.authorizationEndpoint.toString()
-    );
-    const codeVerifier = await this.joseUtility.generateCodeVerifier();
-    // Disable camel case rule because this query requires camel case
+    const redirector = this.redirector;
+    const storage = this.storageUtility;
+
     /* eslint-disable @typescript-eslint/camelcase */
-    const query: { [key: string]: string } = {
-      response_type: "code id_token",
-      redirect_uri: oidcLoginOptions.redirectUrl.toString(),
-      // TODO: the 'webid' scope does not appear in the specification
-      // A question regarding its use has been filed https://github.com/solid/specification
-      scope: "openid webid offline_access",
+    const oidcOptions = {
+      authority: oidcLoginOptions.issuer?.toString(),
       client_id: oidcLoginOptions.client.clientId,
-      code_challenge_method: "S256",
-      code_challenge: await this.joseUtility.generateCodeChallenge(
-        codeVerifier
-      ),
-      state: oidcLoginOptions.sessionId,
+      client_secret: oidcLoginOptions.client.clientSecret,
+      redirect_uri: oidcLoginOptions.redirectUrl?.toString(),
+      post_logout_redirect_uri: oidcLoginOptions.redirectUrl?.toString(),
+      response_type: "code",
+      scope: "openid webid",
+      filterProtocolClaims: true,
+      loadUserInfo: true,
+      code_verifier: true,
     };
     /* eslint-enable @typescript-eslint/camelcase */
-    requestUrl.set("query", query);
-    // TODO: This is inefficent, there should be a bulk
-    await this.storageUtility.setForUser(oidcLoginOptions.sessionId, {
-      codeVerifier,
-      issuer: oidcLoginOptions.issuer.toString(),
-      redirectUri: oidcLoginOptions.redirectUrl.toString(),
-    });
 
-    this.redirector.redirect(requestUrl.toString(), {
-      handleRedirect: oidcLoginOptions.handleRedirect,
-    });
+    const oidcClientLibrary = new OidcClient(oidcOptions);
+
+    oidcClientLibrary
+      .createSigninRequest()
+      .then(async function (req: SigninRequest) {
+        // We use the OAuth 'state' value (which should be crypto-random) as
```

Comment:
Why should it be crypto-random? Isn't the state just a method of preserving arbitrary data over the redirect?

Reply:
It's not just arbitrary data - it needs to be 'unguessable' by an attacker. See section 3.2.3 of the **_OAuth2 In Action_** book.